### ISSUE TITLE
Fix S3900 FP: Don't raise if parameter is captured

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#7060]
### Description

S3900 is falsely raised when the parameter is captured: 

```cs
    public string InsideLambda(string s)
    {
        if (s == null)
        {
            return null;
        }

        Func<string> someFunc = () => s.ToString();

        return s.ToString(); // Noncompliant FP: S3900 "Refactor this method to add a validation of parameter 's'"
    }
```

### Root cause

When we enter the block after the return, `s` has the NotNull constraint as expected. `s` is also in the `captured` list of `LiveVariableAnalysisBase`. Therefore it is not included in the `LiveOut` list:
```cs
        /// <summary>
        /// LiveOut variables are alive when exiting block. They are read in any of it's successors.
        /// </summary>
        public IReadOnlyCollection<ISymbol> LiveOut(TBlock block) =>
            blockLiveOut[block].Except(captured).ToImmutableArray();
```

And that is why it gets removed in `ProcessBranchState`:

https://github.com/SonarSource/sonar-dotnet/blob/0d3caf30c09f7c8e99ab783838e186dd72bf572c/analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs#L165-L167
